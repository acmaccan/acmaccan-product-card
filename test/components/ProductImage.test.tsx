import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('Product Image', () => {
  test('Should render the component with its custom image', () => {
    const wrapper = renderer.create(<ProductImage image={product2.image} />);
    console.log(wrapper.toJSON());
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Should render the component with its default image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
