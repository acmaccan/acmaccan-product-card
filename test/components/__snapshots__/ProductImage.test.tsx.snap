// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image Should render the component with its custom image 1`] = `
<img
  alt="../assets/no-image.jpg"
  className="productImg undefined"
  src="../assets/no-image.jpg"
/>
`;

exports[`Product Image Should render the component with its default image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="../assets/no-image.jpg"
    className="productImg undefined"
    src="../assets/no-image.jpg"
  />
</div>
`;
